core/controllers/topic_editor.py

```diff
@@ -84,6 +89,50 @@ def post(self, topic_id):
         })
 
 
+class TopicEditorQuestionHandler(base.BaseHandler):
+    """Manages the creation of a question and receiving of all question
+    summaries for display in topic editor page.
+    """
+
+    @acl_decorators.can_edit_topic
+    def get(self, topic_id):
+        """Handles GET requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+        topic_domain.Topic.require_valid_topic_id(topic_id)
+
+        topic = topic_services.get_topic_by_id(topic_id)
+        skill_ids = topic.get_all_skill_ids()
+
+        question_summaries = (
+            question_services.get_question_summaries_linked_to_skills(skill_ids)
+        )
+        question_summary_dicts = [
+            summary.to_dict() for summary in question_summaries]
+
+        self.values.update({
+            'question_summary_dicts': question_summary_dicts
+        })
+        self.render_json(self.values)
+
+    @acl_decorators.can_edit_skill
+    def post(self, skill_id):
+        """Handles POST requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+        skill_domain.Skill.require_valid_skill_id(skill_id)
+
+        new_question_id = question_services.get_new_question_id()
+        question = question_domain.Question.create_default_question(
```

Comment:
Wait a sec. This seems wrong?

Like, any question we send in (even for topic managers etc.) should be finalized? Otherwise it's going to be surfaced when users ask for questions for that skill.

Reply:
Done, added new controllers for question skill link and unlink.